File: Demos/Demo 1 Objc/PageMenuDemoStoryboard/PageMenuDemoStoryboard/TestCollectionViewController.m
//
//  TestCollectionViewController.m
//  PageMenuDemoStoryboard
//
//  Created by Jin Sasaki on 2015/06/05.
//  Copyright (c) 2015年 Jin Sasaki. All rights reserved.
//

#import "TestCollectionViewController.h"

@interface TestCollectionViewController ()
@property (nonatomic) NSArray *moodArray;
@property (nonatomic) NSArray *backgroundPhotoNameArray;
@property (nonatomic) NSArray *photoNameArray;
@end

@implementation TestCollectionViewController

static NSString * const reuseIdentifier = @"MoodCollectionViewCell";

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.moodArray = @[@"Relaxed", @"Playful", @"Happy", @"Adventurous", @"Wealthy", @"Hungry", @"Loved", @"Active"];
    self.backgroundPhotoNameArray = @[@"mood1.jpg", @"mood2.jpg", @"mood3.jpg", @"mood4.jpg", @"mood5.jpg", @"mood6.jpg", @"mood7.jpg", @"mood8.jpg"];
    self.photoNameArray = @[@"relax.png", @"playful.png", @"happy.png", @"adventurous.png", @"wealthy.png", @"hungry.png", @"loved.png", @"active.png"];
    
    
    [self.collectionView registerNib:[UINib nibWithNibName:@"MoodCollectionViewCell" bundle:nil] forCellWithReuseIdentifier:reuseIdentifier];
    
}


#pragma mark <UICollectionViewDataSource>

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}


- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return 8;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    MoodCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];

    // Configure the cell
    cell.moodTitleLabel.text = self.moodArray[indexPath.row];
    cell.backgroundImageView.image = [UIImage imageNamed: self.backgroundPhotoNameArray[indexPath.row]];
    cell.moodIconImageView.image = [UIImage imageNamed: self.photoNameArray[indexPath.row]];
    
    
    return cell;
}
@end

File: Demos/Demo 6/PageMenuDemoMenuItemWidthBasedOnTitleText/PageMenuDemoMenuItemWidthBasedOnTitleText/TestCollectionViewController.swift
//
//  TestCollectionViewController.swift
//  NFTopMenuController
//
//  Created by Niklas Fahl on 12/17/14.
//  Copyright (c) 2014 Niklas Fahl. All rights reserved.
//

import UIKit

let reuseIdentifier = "MoodCollectionViewCell"

class TestCollectionViewController: UICollectionViewController {
    
    var moodArray : [String] = ["Relaxed", "Playful", "Happy", "Adventurous", "Wealthy", "Hungry", "Loved", "Active"]
    var backgroundPhotoNameArray : [String] = ["mood1.jpg", "mood2.jpg", "mood3.jpg", "mood4.jpg", "mood5.jpg", "mood6.jpg", "mood7.jpg", "mood8.jpg"]
    var photoNameArray : [String] = ["relax.png", "playful.png", "happy.png", "adventurous.png", "wealthy.png", "hungry.png", "loved.png", "active.png"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.collectionView!.registerNib(UINib(nibName: "MoodCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: reuseIdentifier)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: UICollectionViewDataSource

    override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
        //#warning Incomplete method implementation -- Return the number of sections
        return 1
    }


    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        //#warning Incomplete method implementation -- Return the number of items in the section
        return 8
    }

    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
        let cell : MoodCollectionViewCell = collectionView.dequeueReusableCellWithReuseIdentifier(reuseIdentifier, forIndexPath: indexPath) as! MoodCollectionViewCell
    
        // Configure the cell
        cell.backgroundImageView.image = UIImage(named: backgroundPhotoNameArray[indexPath.row])
        cell.moodTitleLabel.text = moodArray[indexPath.row]
        cell.moodIconImageView.image = UIImage(named: photoNameArray[indexPath.row])
    
        return cell
    }
}
